device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -475,7 +476,6 @@ public void registerDeviceClients(Iterable<DeviceClient> deviceClients, long tim
                 }
                 else
                 {
-                    this.multiplexedDeviceClients.put(deviceClientToRegister.getConfig().getDeviceId(), deviceClientToRegister);
```

Comment:
Putting the client in the map of multiplexed devices here was premature since the registration hasn't actually happened yet. The client has only been validated to ensure that it has the expected protocol, hostname, etc.